pkg/chunked/storage_linux.go

```diff
@@ -949,6 +978,9 @@ func (c *chunkedDiffer) retrieveMissingFiles(stream ImageSourceSeekable, dirfd i
 		if _, ok := err.(ErrBadRequest); ok {
 			// If the server cannot handle at least 64 chunks in a single request, just give up.
 			if len(chunksToRequest) < 64 {
+				if c.canFallback {
```

Comment:
Isn’t this too late? If the registry doesn’t support the feature, `readZstdChunkedManifest` / `readEstargzChunkedManifest` is going to fail, and _that_ needs to be annotated with the error type for fallback to be possible at all.

This `retrieveMissingFiles` is only called from `Differ.ApplyDiff`, that’s all after `GetDiffer` returns. (Are you proposing that `DifferApplyDiff` should _also_ have the special error type to trigger fallback? That would be possible but seems unnecessary to me.)

Reply:
this entire code block is pure theoretical, as I don't remember ever hitting this case.  Should we just drop the `if len(chunksToRequest) < 64 {` and not have to worry about it?

Reply:
> Isn’t this too late? If the registry doesn’t support the feature, `readZstdChunkedManifest` / `readEstargzChunkedManifest` is going to fail, and _that_ needs to be annotated with the error type for fallback to be possible at all.

in theory the registry could support less than 64 ranges in a request.  With `readZstdChunkedManifest / readEstargzChunkedManifest` we request only one

Reply:
I added a patch to drop this condition

Reply:
> Should we just drop the `if len(chunksToRequest) < 64 {` and not have to worry about it?

I have no opinion, I’m fine with dropping that logic … but …